alacritty_terminal/src/term/search.rs

```diff
@@ -0,0 +1,390 @@
+            for b in 0..cell.c.len_utf8() {
+                // Since we get the state from the DFA, it doesn't need to be checked
+                state = unsafe { dfa.next_state_unchecked(state, buf[b]) };
+            }
+
+            // Handle regex state changes
+            if dfa.is_match_or_dead_state(state) {
+                if dfa.is_dead_state(state) {
+                    if regex_match.is_some() {
+                        return regex_match;
+                    }
+
+                    state = dfa.start_state();
+                } else {
+                    regex_match = Some(point);
+                }
+            }
+
+            // Advance iterator
+            let new_cell = match iter_fn(iter) {
+                Some(&cell) => cell,
+                None => break,
+            };
+            let last_point = mem::replace(&mut point, iter.point());
+            let last_cell = mem::replace(&mut cell, new_cell);
+
+            // Reset on line breaks
+            if (last_point.col == last_col && !last_cell.flags.contains(Flags::WRAPLINE))
+                || (last_point.col == Column(0)
+                    && point.col == last_col
+                    && !cell.flags.contains(Flags::WRAPLINE))
+            {
+                if regex_match.is_some() {
+                    return regex_match;
+                }
+
+                state = dfa.start_state();
+            }
+        }
+
+        regex_match
+    }
+
+    pub fn bracket_search(&self, point: Point<usize>) -> Option<Point<usize>> {
+        let start_char = self.grid[point.line][point.col].c;
+
+        // Find the matching bracket we're looking for
+        let (forwards, end_char) = BRACKET_PAIRS.iter().find_map(|(open, close)| {
+            if open == &start_char {
+                Some((true, *close))
+            } else if close == &start_char {
+                Some((false, *open))
+            } else {
+                None
+            }
+        })?;
+
+        let mut iter = self.grid.iter_from(point);
+
+        // For every character match that equals the starting bracket, we
+        // ignore one bracket of the opposite type.
+        let mut skip_pairs = 0;
+
+        loop {
+            // Check the next cell
+            let cell = if forwards { iter.next() } else { iter.prev() };
+
+            // Break if there are no more cells
+            let c = match cell {
+                Some(cell) => cell.c,
+                None => break,
+            };
+
+            // Check if the bracket matches
+            if c == end_char && skip_pairs == 0 {
+                return Some(iter.point());
+            } else if c == start_char {
+                skip_pairs += 1;
+            } else if c == end_char {
+                skip_pairs -= 1;
+            }
+        }
+
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::index::{Column, Line};
+    use crate::term::test::mock_term;
```

Comment:
`grep -r mock_term alacritty_terminal` yields no results (other than this file).  Also there's a typo on `tests`.

Reply:
This PR is a WIP and was accidentally half reverted.